### DISCUSSION TITLE
EIP712 iOS Not Working

### DISCUSSION THREAD

**Discussion by trevormil** [trustwallet/wallet-core#3129] [Q&A]
As described in https://github.com/trustwallet/wallet-core/discussions/3049, Trust Wallet is having trouble signing EIP712 signed data on iOS. 

I am using Web3Modal and wagmi. When I call signTypedData() on desktop, it works fine. When I attempt to call it with Trust Wallet iOS, it does not produce the same signature.

Here is my _app.tsx.

```ts
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { Web3Modal } from '@web3modal/react'
import { configureChains, createClient, WagmiConfig } from 'wagmi'
import { mainnet } from 'wagmi/chains'

const chains = [mainnet]
const projectId = 'febf8d9986a2cd637fa4004338dad39b'

const { provider } = configureChains(chains, [w3mProvider({ projectId })])
const wagmiClient = createClient({
    autoConnect: true,
    connectors: w3mConnectors({ projectId, version: 1, chains }),
    provider,
})
const ethereumClient = new EthereumClient(wagmiClient, chains)
const queryClient = new QueryClient()
```
and here is my sign function:

```ts
import { disconnect as disconnectWeb3, signMessage, signTypedData } from "@wagmi/core";
...
...
const sig = await signTypedData({
    value: txn.eipToSign.message,
    types: txn.eipToSign.types,
    domain: txn.eipToSign.domain
});
```

**Comment by trevormil** [comment]
Android works as well

**Comment by Milerius** [comment]
> As described in #3049, Trust Wallet is having trouble signing EIP712 signed data on iOS.
> 
> I am using Web3Modal and wagmi. When I call signTypedData() on desktop, it works fine. When I attempt to call it with Trust Wallet iOS, it does not produce the same signature.
> 
> Here is my _app.tsx.
> 
> ```ts
> import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
> import { Web3Modal } from '@web3modal/react'
> import { configureChains, createClient, WagmiConfig } from 'wagmi'
> import { mainnet } from 'wagmi/chains'
> 
> const chains = [mainnet]
> const projectId = 'febf8d9986a2cd637fa4004338dad39b'
> 
> const { provider } = configureChains(chains, [w3mProvider({ projectId })])
> const wagmiClient = createClient({
>     autoConnect: true,
>     connectors: w3mConnectors({ projectId, version: 1, chains }),
>     provider,
> })
> const ethereumClient = new EthereumClient(wagmiClient, chains)
> const queryClient = new QueryClient()
> ```
> 
> and here is my sign function:
> 
> ```ts
> import { disconnect as disconnectWeb3, signMessage, signTypedData } from "@wagmi/core";
> ...
> ...
> const sig = await signTypedData({
>     value: txn.eipToSign.message,
>     types: txn.eipToSign.types,
>     domain: txn.eipToSign.domain
> });
> ```

Doesn't seem to be related to wallet-core library, converting to a discussion.
